### PR TITLE
Support `deviceFrame:` parameter in `@Image` and `@Video` directives

### PR DESCRIPTION
**Pull request by ethan-kusters**
Bug/issue #, if applicable: rdar://102183073

## Summary

When developing UI libraries or apps,
it can be useful to show previews of how things look
when rendered on a device, like a phone or a tablet.
This new experimental feature is designed to allow authors to wrap
images and videos in a frame, making it easier to create complex assets
and maintain them overtime.

Both `@Image` and `@Video` gain a `deviceFrame` parameter that
associates them with a device frame defined in the DocC catalog’s
`theme-settings.json` file.

This change is described on the Swift forums here:
https://forums.swift.org/t/support-for-device-frames/62151

## Dependencies

DocC-Render PR TBD

### Test that `deviceFrame` works as expected

1. Add an `@Image` or `@Video` directive to a DocC article:

    ```md
    # My Great Article
    
    This is a great one!
    
    With a lot of great content.
    
    @Image(source: "sloth", deviceFrame: phone)
    ```

2. Confirm that the emitted RenderJSON references the `"phone"` device frame

### Test that `deviceFrame` is excluded from documentation

This is an experimental feature that shouldn't be advertised in documentation yet.

1. `swift run generate-symbol-graph`
2. `bin/preview-docs DocC`
3. <http://localhost:8080/documentation/docc/image>
4. Confirm that `@Image` and `@Video` declarations don't include `deviceFrame`:

    Screenshot 2023-01-11 at 1 57 54 PM

## Checklist

Make sure you check off the following items. If they cannot be completed, provide a reason.

- [x] Added tests
- [x] Ran the `./bin/test` script and it succeeded
- [x] Updated documentation if necessary

### CODE REVIEW

File: Sources/SwiftDocC/Model/Rendering/Content/RenderContentMetadata.swift [hunk 1]
```diff
@@ -18,6 +18,8 @@ public struct RenderContentMetadata: Equatable, Codable {
     public var title: String?
     /// An optional custom abstract.
     public var abstract: [RenderInlineContent]?
+    /// An optional identifier for the device frame that should wrap this element.
```
Comment (d-ronnqvist) [inline Sources/SwiftDocC/Model/Rendering/Content/RenderContentMetadata.swift:21]: Should we add the same experimental-feature mention to this doc comment that we have for the `DirectiveArgumentWrapped ` property?
```
/// This is an experimental feature – any device frame specified here
/// must be defined in the `theme-settings.json` file of the containing DocC catalog.
```

**Review by d-ronnqvist — APPROVED** [review]
This looks good to me as an experimental feature

**Comment by ethan-kusters** [comment]
@swift-ci please test

**Comment by ethan-kusters** [comment]
@swift-ci please test

**Comment by ethan-kusters** [comment]
@swift-ci please test

**Pull request merged**